File: perf/bench/matmul/matmul.cpp
#include <cstdio>

static long long a[200][200], b[200][200], c[200][200];

int main() {
    for (int i = 0; i < 200; i++)
        for (int j = 0; j < 200; j++) {
            a[i][j] = (i + j) % 7;
            b[i][j] = ((long long)i * j) % 5;
            c[i][j] = 0;
        }
    for (int rep = 0; rep < 20; rep++)
    for (int i = 0; i < 200; i++)
        for (int k = 0; k < 200; k++) {
            long long aik = a[i][k];
            for (int j = 0; j < 200; j++)
                c[i][j] += aik * b[k][j];
        }
    long long sum = 0;
    for (int i = 0; i < 200; i++) sum += c[i][i];
    std::printf("%lld\n", sum);
    return 0;
}

File: perf/bench/matmul/matmul.rs
fn main() {
    let mut a = vec![[0i64; 200]; 200];
    let mut b = vec![[0i64; 200]; 200];
    let mut c = vec![[0i64; 200]; 200];
    for i in 0..200usize {
        for j in 0..200usize {
            a[i][j] = ((i + j) % 7) as i64;
            b[i][j] = ((i * j) % 5) as i64;
            c[i][j] = 0;
        }
    }
    for _rep in 0..20 {
    for i in 0..200usize {
        for k in 0..200usize {
            let aik = a[i][k];
            for j in 0..200usize {
                c[i][j] += aik * b[k][j];
            }
        }
    }
    }
    let mut sum: i64 = 0;
    for i in 0..200usize {
        sum += c[i][i];
    }
    println!("{}", sum);
}
